### PR TITLE
[web] Check for cached browsers on LUCI

### PR DESCRIPTION
**Pull request by hterkelsen**
Engine-side change for https://github.com/flutter/flutter/issues/115404

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/dev/chrome_installer.dart [hunk 1]
```diff
@@ -33,6 +35,17 @@ Future<BrowserInstallation> getOrInstallChrome(
 }) async {
   infoLog ??= io.stdout;
 
+  // When running on LUCI, if we specify the "chrome_and_driver" dependency,
+  // then the bot will download Chrome from CIPD and place it in a cache and
+  // set the environment variable CHROME_EXECUTABLE.
+  if (io.Platform.environment.containsKey(_chromeExecutableVar)) {
+    infoLog.writeln('Using Chrome from cache: ${io.Platform.environment[_chromeExecutableVar]}');
```
Comment (yjbanov) [inline lib/web_ui/dev/chrome_installer.dart:42]: nit: I'd replace the word "cache" with `$_chromeExecutableVar variable` as we could use it for other things, such as testing WasmGC on Chrome Canary and such.

File: lib/web_ui/dev/firefox_installer.dart [hunk 2]
```diff
@@ -36,6 +38,17 @@ Future<BrowserInstallation> getOrInstallFirefox(
 
   infoLog ??= io.stdout;
 
+  // When running on LUCI, if we specify the "firefox" dependency, then the
+  // bot will download Firefox from CIPD and place it in a cache and set the
+  // environment variable FIREFOX_EXECUTABLE.
+  if (io.Platform.environment.containsKey(_firefoxExecutableVar)) {
+    infoLog.writeln('Using Firefox from cache: ${io.Platform.environment[_firefoxExecutableVar]}');
```
Comment (yjbanov) [inline lib/web_ui/dev/firefox_installer.dart:45]: nit: I'd replace the word "cache" with `$_firefoxExecutableVar variable` for the same reason as Chrome above.

**Review by yjbanov — APPROVED** [review]

File: lib/web_ui/dev/firefox_installer.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/dev/firefox_installer.dart]: Done.

File: lib/web_ui/dev/chrome_installer.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/dev/chrome_installer.dart]: Done.

**Pull request merged**